vignettes/blueprint/module_encapsulation.Rmd

```diff
@@ -0,0 +1,25 @@
+---
+title: "Module Encapsulation"
+author: "NEST CoreDev"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Module Encapsulation}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` leverages the [Shiny module](https://rstudio.github.io/shiny/reference/moduleServer.html) concept to enable seamless communication between the framework and `teal` modules, while also promoting encapsulation.
+
+`teal` modules heavily depends on this core Shiny feature and is a powerful extension that allows for a seamless
+
+## Benefits
+
+The shiny module technique provides an efficient and structured approach when developing large shiny apps.
+By incorporating the shiny module concept into `teal`, developers can break down complex functionalities into smaller, more manageable modules, encapsulating specific functionality within each module.
+
+This encapsulation ensures that the variables, functions, and logic within a module are self-contained and independent from other parts of the application.
+As a result, developers can focus on building self-contained components within each module, making it easier to understand, test, and enhance different aspects of the application independently.
```

Comment:
These two paragraphs summarize the concept of modular `shiny` apps as such and say nothing much about `teal` specifically. I think they are redundant.

Reply:
I still feel that this gives a good context for new users that are now familiar with shiny module.
Let's keep it for now.

Reply:
The official article on modules is linked a few lines above. If we are to summarize it, at least let's not phrase is as something that `teal` introduces. Modular apps are not a benefit of `teal`, they are a benefit _to_ `teal`.

Reply:
I see your point.

What do you think about this?

By implementing the modular app technique from the shiny module into the creation of the teal module, several benefits are realized:

1. Streamlined Maintenance
The development of the teal module becomes more manageable, as it can function independently from the teal framework. This separation allows developers to maintain the module with ease. This approach has been successfully applied in R packages dedicated to teal module development, such as teal.modules.general and teal.modules.clinical.

2. Enhanced Focus on Output
teal module developers can concentrate solely on refining parameters or encoding, and output aspects (such as data summarization and visualization) without the need to concern themselves with the intricacies of the teal framework. When developed correctly, the module seamlessly integrates with teal.

3. Facilitated Collaboration
teal module development becomes an accessible entry point for developers interested in collaborating. This approach encourages user collaboration for the improvement of teal modules, as developers gain a deeper understanding of the mechanics of the teal framework.